cli/flox/src/commands/containerize/macos_containerize_proxy.rs

```diff
@@ -1,21 +1,104 @@
 use std::convert::Infallible;
 use std::path::PathBuf;
 
+use dirs::home_dir;
+use flox_rust_sdk::flox::FLOX_VERSION;
 use flox_rust_sdk::providers::container_builder::{ContainerBuilder, ContainerSource};
 
-#[allow(unused)]
-struct ContainerizeProxy {
+use super::Runtime;
+
+const FLOX_FLAKE: &str = "github:flox/flox";
+const FLOX_PROXY_IMAGE: &str = "ghcr.io/flox/flox";
+
+#[derive(Debug, Clone)]
+pub(crate) struct ContainerizeProxy {
     environment_path: PathBuf,
+    container_runtime: Runtime,
+}
+
+impl ContainerizeProxy {
+    pub(crate) fn new(environment_path: PathBuf, container_runtime: Runtime) -> Self {
+        Self {
+            environment_path,
+            container_runtime,
+        }
+    }
 }
 
 impl ContainerBuilder for ContainerizeProxy {
     type Error = Infallible;
 
     fn create_container_source(
         &self,
-        _name: impl AsRef<str>,
-        _tag: impl AsRef<str>,
+        _name: impl AsRef<str>, // Inferred from `self.environment_path`
+        tag: impl AsRef<str>,
     ) -> Result<ContainerSource, Self::Error> {
-        todo!("🚧 MacOS container builder in construction 🚧")
+        let env_mount = "/flox_env";
+        let home_mount = "/flox_home";
+
+        // Inception L1: Container runtime args.
+        let mut command = self.container_runtime.to_command();
+        command.arg("run");
+        command.arg("--rm");
+        command.args([
+            "--mount",
+            &format!(
+                "type=bind,source={},target={}",
+                self.environment_path.to_string_lossy(),
+                env_mount
+            ),
+        ]);
+
+        // Honour config from the user's home directory on their host machine if
+        // available.
+        if let Some(home_dir) = home_dir() {
+            command.args(["--env", &format!("HOME={}", home_mount)]);
+            command.args([
+                "--mount",
+                &format!(
+                    "type=bind,source={},target={}",
+                    home_dir.to_string_lossy(),
+                    home_mount
+                ),
+            ]);
+        }
+
```

Comment:
suggestion: disable metrics iff disabled by the host

Reply:
Done (will rebase fixup before merge)